src/backend/distributed/planner/shard_pruning.c

```diff
@@ -756,6 +756,17 @@ AddPartitionKeyRestrictionToInstance(ClauseWalkerContext *context, OpExpr *opCla
 		}
 	}
 
+	if (constantClause->constisnull)
+	{
+		/*
+		 * We cannot do pruning for NULL values, so we note this as a
+		 * restriction we don't grok.
```

Comment:
grok?

Reply:
> grok


that's a word that @jasonmp85 used in the few lines above, just copy-pasted :D I'll update both

Reply:
Roughly "understand" or "comprehend". It's slangy, sorry, but at least fairly prevalent among anglophone-country techies…

>  1. (_transitive, slang_) To understand (something) intuitively, to know (something) without having to think intellectually.
>  2. (_transitive, slang_) To fully and completely understand something in all of its details and intricacies.

[Wiktionary entry](https://en.wiktionary.org/wiki/grok). My first instinct is it originated in Heinlein's _Stranger in a Strange Land_, which was apparently correct.

> Grok is used mainly by the geek subculture, though it was heavily used by the counterculture of the 1960s, as evidenced by its repeated appearance in Tom Wolfe's _The Electric Kool-Aid Acid Test._